dev-infrastructure/templates/aks-development.bicep

```diff
@@ -123,6 +90,66 @@ resource aks_pod_nsg 'Microsoft.Network/networkSecurityGroups@2023-09-01' = {
   location: location
 }
 
+resource aks_keyvault 'Microsoft.KeyVault/vaults@2023-07-01' = {
+  location: location
+  name: take('aks-kv-${uniqueString(currentUserId)}', 24)
+  tags: {
+    resourceGroup: resourceGroup().name 
+  }
+  properties: {
+    enableRbacAuthorization: true
+    enabledForDeployment: false
+    enabledForDiskEncryption: false
+    enabledForTemplateDeployment: false
+    networkAcls: {
+      bypass: 'AzureServices'
+      defaultAction: 'Deny'
+      ipRules: [
+        {
+          // TODO: restrict in higher environments
+          value: '0.0.0.0/0'
+        }
+      ]
+    }
+    // TODO: disabled in higher environments
```

Comment:
Are we going to do these in this PR?  Or later?

Reply:
Later - it's not trivial to setup a private Keyvault in dev and just trying to get something out that's usable for testing